lib/auth/auth_with_roles.go

```diff
@@ -783,30 +788,38 @@ func (a *ServerWithRoles) GetNodes(ctx context.Context, namespace string, opts .
 }
 
 // ListResources returns a paginated list of resources filtered by user access.
-func (a *ServerWithRoles) ListResources(ctx context.Context, req proto.ListResourcesRequest) ([]types.Resource, string, error) {
+func (a *ServerWithRoles) ListResources(ctx context.Context, req proto.ListResourcesRequest) ([]types.ResourceWithLabels, string, error) {
 	if err := req.CheckAndSetDefaults(); err != nil {
 		return nil, "", trace.Wrap(err)
 	}
 
 	limit := int(req.Limit)
-
-	var resourceKind string
 	switch req.ResourceType {
-	case types.KindDatabaseServer:
-		resourceKind = types.KindDatabaseServer
-	case types.KindAppServer:
-		resourceKind = types.KindAppServer
+	case types.KindDatabaseServer,
+		types.KindAppServer,
+		types.KindNode,
+		types.KindKubeService:
```

Comment:
Sorry gabriel, I don't think I was clear on my earlier comment. What I was suggesting was in addition to these kinds, we should also add `KindApplication` and `KindDatabase` that returns list of apps and dbs instead of servers. It will be similar to desktops. When it's time to add desktops here, we will need both `KindWindowsDesktopService` which is equiv to `GetWindowsDesktopService` and `KindWindowsDesktop` -> `GetWindowsDesktop`. I would leave label matching as it was before, unless we got confirmation that merging of labels is the behavior we want